opslevel/datasource_opslevel_system.go

```diff
@@ -1,63 +1,114 @@
-// 	d.Set("description", resource.Description)
-// 	d.Set("owner", resource.Owner.Id())
-// 	d.Set("domain", resource.Parent.Id)
-
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+// Ensure SystemDataSource implements DataSourceWithConfigure interface
+var _ datasource.DataSourceWithConfigure = &SystemDataSource{}
+
+func NewSystemDataSource() datasource.DataSource {
+	return &SystemDataSource{}
+}
+
+// SystemDataSource manages a System data source.
+type SystemDataSource struct {
+	CommonDataSourceClient
+}
+
+// SystemDataSourceModel describes the data source data model.
+type SystemDataSourceModel struct {
+	Aliases     types.List   `tfsdk:"aliases"`
+	Description types.String `tfsdk:"description"`
+	Domain      types.String `tfsdk:"domain"`
+	Id          types.String `tfsdk:"id"`
+	Identifier  types.String `tfsdk:"identifier"`
+	Name        types.String `tfsdk:"name"`
+	Owner       types.String `tfsdk:"owner"`
+}
+
+func NewSystemDataSourceModel(ctx context.Context, system opslevel.System, identifier types.String) (SystemDataSourceModel, diag.Diagnostics) {
+	aliases, diags := types.ListValueFrom(ctx, types.StringType, system.Aliases)
+	return SystemDataSourceModel{
+		Aliases:     aliases,
+		Description: types.StringValue(system.Description),
+		Domain:      types.StringValue(string(system.Parent.Id)),
+		Id:          types.StringValue(string(system.Id)),
+		Identifier:  identifier,
+		Name:        types.StringValue(system.Name),
+		Owner:       types.StringValue(string(system.Owner.Id())),
+	}, diags
+}
+
+func (sys *SystemDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_system"
+}
+
+func (sys *SystemDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "System data source",
+
+		Attributes: map[string]schema.Attribute{
+			"aliases": schema.ListAttribute{
+				ElementType:         types.StringType,
+				MarkdownDescription: "All of the aliases attached to the System.",
+				Computed:            true,
+			},
+			"description": schema.StringAttribute{
+				MarkdownDescription: "The description of the System.",
+				Computed:            true,
+			},
+			"domain": schema.StringAttribute{
+				MarkdownDescription: "ID of the parent domain of the System.",
+				Computed:            true,
+			},
+			"id": schema.StringAttribute{
+				Description: "Terraform specific identifier.",
```

Comment:
```suggestion
		        	Description: "The ID of this resource."
```
I need to go back and update the id description for `Domain` and `Filter`, they have this same description too...